src/coreComponents/dataRepository/DataContext.hpp

```diff
@@ -223,4 +223,23 @@ struct GEOS_FMT_NS::formatter< geos::dataRepository::DataContext > : GEOS_FMT_NS
   }
 };
 
+// The following workaround is needed to fix compilation with NVCC on some machines.
+// The standard definition of the has_const_formatter check of fmt fails due to a compiler bug, see the issue below: 
+// https://github.com/fmtlib/fmt/issues/2746
+// The workaround was originally implemented in fmt:
+// https://github.com/fmtlib/fmt/commit/70de324aa801eaf52e94c402d526a3849797c620
+// but later removed: 
+// https://github.com/fmtlib/fmt/commit/466e0650ec2d153d255a40ec230eb77d7f1c3334
+// This workaround provides a specialization of the const formatter check for the DataContext object.
+// The formatter is defined within this file, and therefore the check is not needed.
+// The scope of the check override is as small as possible to solve the current issue.
+#ifdef GEOS_USE_CUDA
```

Comment:
As the compilation on Pecan GPU works, the problem doesn't seem to come from NVCC. It seems to come from the Power9 compilation target.
If no define are declared by the compiler, it would be possible to add a specific define for this workaround (`FMT_CONST_FORMATTER_WORKAROUND` ?)

Reply:
I think the workaround tag is a good idea--it will allow to activate/deactivate it on a case-by-case basis.